tests/examples/shims/keygrabber.lua

```diff
@@ -1,11 +1,60 @@
 
-local keygrabber = {
-    run = function() end,
-    stop = function() end,
-    is_running = function() return false end,
+local keygrabber = nil
+
+local function run(grabber)
+    rawset(keygrabber, "_current_grabber", grabber)
+end
+
+local function stop()
+    keygrabber._current_grabber = nil
+end
+
+keygrabber = {
+    _run             = run,
+    stop             = stop,
+    isrunning        = function() return keygrabber._current_grabber ~= nil end,
+    _current_grabber = nil
 }
 
+function keygrabber.set_newindex_miss_handler(h)
+    rawset(keygrabber, "_ni_handler", h)
+end
+
+function keygrabber.set_index_miss_handler(h)
+    rawset(keygrabber, "_i_handler", h)
+end
+
+function keygrabber.set_call_handler(h)
+    rawset(keygrabber, "_c_handler", h)
+end
```

Comment:
What's that? I don't think the C API has this function.

Reply:
It is added by one of the other commit